bundles/org.eclipse.swt/Eclipse SWT/common/org/eclipse/swt/graphics/ISVGRasterizer.java

```diff
@@ -0,0 +1,27 @@
+package org.eclipse.swt.graphics;
+
+import java.awt.image.*;
+import java.io.*;
+
+/**
+ * Defines the interface for an SVG rasterizer, responsible for converting SVG
+ * data into rasterized images.
+ *
+ * @since 3.129
+ */
+public interface ISVGRasterizer {
+
+	/**
+	 * Rasterizes an SVG image from the provided byte array, using the specified
+	 * zoom factor.
+	 *
+	 * @param bytes         the SVG image as a byte array.
+	 * @param scalingFactor the scaling ratio e.g. 2.0 for doubled size.
+	 * @return a {@link BufferedImage} containing the rasterized image, or
+	 *         {@code null} if the input is not a valid SVG file or cannot be
+	 *         processed.
+	 * @throws IOException if an error occurs while reading the SVG data.
+	 */
+	public BufferedImage rasterizeSVG(byte[] bytes, float scalingFactor) throws IOException;
```

Comment:
Why not use `org.eclipse.swt.graphics.ImageData` here?

Using AWT class `java.awt.image.BufferedImage` in public SWT API is probably a bad idea, isn't it?
This means that all implementations must either use AWT for rendering, or if using some native library then an additional conversion from native image data to AWT BufferedImage would be necessary.

Reply:
You are right that I could move the conversion you mentioned in your second comment into the `rasterizeSVG` method and then directly return `org.eclipse.swt.graphics.ImageData`.

Reply:
ImageData has different implementations but they all expose the same API so you don't have to choose, rather stick to the public API and SWT uses the implementation for the given OS.  What do you need FileFormat for ?

Reply:
I changed this in ffa17e4b4f65481d700b9d7fc27c305934166080 like @DevCharly proposed as it fixed the conversion issue and AWT does not need to be used in SWT.